### ISSUE TITLE
azureml service wrong Keyvault Configuration

### ISSUE THREAD

**Issue by chboudry** [microsoft/AzureTRE#1654]
With the default AML workspace is not able to access keyvault which will result in a "forbidden" error while trying to list datastore or create new ones.

To fix this : 
Key vault is missing the "Allow trusted Microsoft services to bypass this firewall" in the network blade of the Keyvault.

in Terraform it translates by adding a bypass property with a value of "AzureServices" on the network_acls block
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/key_vault

**Comment by chboudry** [comment]
Hello @dram1964, in the current version of TRE, the network settings of the storage account for the workspace already has this configuration.

**Comment by dram1964** [comment]
Hi @chboudry - sorry my mistake. Yes the storage does already have this already. Please ignore - the fix was implemented in the keyvault as you said.

**Comment by marrobi** [comment]
We need to fix this be either adding the setting on the base workspace KeyVault, or deploying a dedicated KV for Azure ML. 

Both would only be accessible from within the workspace, so question is there a risk of data exfiltration if enable access to the trusted services listed here: https://docs.microsoft.com/en-us/azure/key-vault/general/overview-vnet-service-endpoints#trusted-services

@chboudry thoughts?

The other option is have a specific workspace template for Azure ML given some of the data exfiltration challenges.

**Comment by chboudry** [comment]
@marrobi As trusted service config is already used on storage account level it would make sense to have a similar configuration for the keyvault.

**Comment by marrobi** [comment]
Good point, if its set in one, then makes sense to be set in the other.

Should also document the fact that this is set.

**Comment by dram1964** [comment]
I've manually applied that fix in the version of the AzureTre that I'm using and it also seems to have fixed Issue #1255 
For reference I've changed
```
    bypass         = "None"
```
to 
```
    bypass         = "AzureServices"
```
in templates > workspaces > base > terraform > keyvault.tf
